### DISCUSSION TITLE
How to deal with vouchers

### DISCUSSION THREAD

**Discussion by engelhuber** [stephanstapel/ZUGFeRD-csharp#376] [General]
Hi. I'm trying to get a valid ZUGFeRD XML containing a voucher with 0% VAT.

I've added the 50€ voucher as **SpecifiedTradeAllowanceCharge** like the following

```
      <ram:SpecifiedTradeAllowanceCharge>
        <ram:ChargeIndicator>
          <udt:Indicator>false</udt:Indicator>
        </ram:ChargeIndicator>
        <ram:BasisAmount>0.00</ram:BasisAmount>
        <ram:ActualAmount>50.00</ram:ActualAmount>
        <ram:Reason>Gutschein</ram:Reason>
        <ram:CategoryTradeTax>
          <ram:TypeCode>VAT</ram:TypeCode>
          <ram:CategoryCode>Z</ram:CategoryCode>
          <ram:RateApplicablePercent>0.00</ram:RateApplicablePercent>
        </ram:CategoryTradeTax>
      </ram:SpecifiedTradeAllowanceCharge>
```

I also added an **ApplicableTradeTax** for the zero vat category:

```
      <ram:ApplicableTradeTax>
        <ram:CalculatedAmount>0.00</ram:CalculatedAmount>
        <ram:TypeCode>VAT</ram:TypeCode>
        <ram:BasisAmount>6.35</ram:BasisAmount>
        <ram:CategoryCode>Z</ram:CategoryCode>
        <ram:RateApplicablePercent>0.00</ram:RateApplicablePercent>
      </ram:ApplicableTradeTax>
```

but I can't get rid of the follwing error. Tried several combinations of the BasisAmount. It doesn't work.

_[BR-Z-08]-In a VAT breakdown (BG-23) where VAT category code (BT-118) is "Zero rated"the VAT category taxable amount (BT-116) shall equal the sum of Invoice line net amount(BT-131) minus the sum of Document level allowance amounts (BT-92) plus the sum ofDocument level charge amounts (BT-99) where the VAT category codes (BT-151, BT-95,BT-102) are "Zero rated"._

Net invoice amount: 56,35€
VAT amount: 10,71€
Total incl. VAT: 67,06€ (-50€ voucher)

**Comment by DaveHalma** [comment]
Hi engelhuber.
Try changing the TaxCategory Code. Maybe to "E"?

**Comment by engelhuber** [comment]
> Hi engelhuber. Try changing the TaxCategory Code. Maybe to "E"?

thanks for your help, but this brings nearly the same error ;)

_[BR-E-08]-In a VAT breakdown (BG-23) where the VAT category code (BT-118) is "Exemptfrom VAT" the VAT category taxable amount (BT-116) shall equal the sum of Invoiceline net amounts (BT-131) minus the sum of Document level allowance amounts (BT-92)plus the sum of Document level charge amounts (BT-99) where the VAT category codes(BT-151, BT-95, BT-102) are "Exempt from VAT"._

I had to add an **ExemptionReasonCode** for the CategoryCode = E

```
      <ram:ApplicableTradeTax>
        <ram:CalculatedAmount>0.00</ram:CalculatedAmount>
        <ram:TypeCode>VAT</ram:TypeCode>
        <ram:BasisAmount>0.00</ram:BasisAmount>
        <ram:CategoryCode>E</ram:CategoryCode>
	<ram:ExemptionReasonCode>VATEX-EU-AE</ram:ExemptionReasonCode>
        <ram:RateApplicablePercent>0.00</ram:RateApplicablePercent>
      </ram:ApplicableTradeTax>	  
	  
      <ram:SpecifiedTradeAllowanceCharge>
        <ram:ChargeIndicator>
          <udt:Indicator>false</udt:Indicator>
        </ram:ChargeIndicator>
        <ram:BasisAmount>67.06</ram:BasisAmount>
        <ram:ActualAmount>50.00</ram:ActualAmount>
        <ram:Reason>Gutschein</ram:Reason>
        <ram:CategoryTradeTax>
          <ram:TypeCode>VAT</ram:TypeCode>
          <ram:CategoryCode>E</ram:CategoryCode>
          <ram:RateApplicablePercent>0.00</ram:RateApplicablePercent>
        </ram:CategoryTradeTax>
      </ram:SpecifiedTradeAllowanceCharge>
```

The problem is, I have no invoice lines where the VAT category code is "Exempt from VAT". I only added the voucher as allowence / discount.
At the beginning I tried to add the voucher as a negative invoice line. But this is also not allowed. 

I'm wondering that I can't find any discussions about ZUGFeRD incl. vouchers :(

**Comment by jochenkirstaetter** [comment]
Hi @engelhuber 

Has the `Type` been set to `CreditNote`? 
Maybe it's related to the invoice type how the validation rules are applied?

**Reply by engelhuber** [reply]
> Hi @engelhuber
> 
> Has the `Type` been set to `CreditNote`? Maybe it's related to the invoice type how the validation rules are applied?

Hi @jochenkirstaetter you mean the TypeCode in the header? No it's 380 = Invoice

Actually it is an invoice, which is partly payed with a voucher.

**Comment by IsThatUniqueEnough** [comment]
Probably a question for your tax adviser... For me a voucher is not an allowance. And I wouldn't flag an amount as zero rated or exempted from tax unless there's an applicable law justifying to do so.

IMHO a voucher is a payment mean, so I'd likely add its amount to ram:TotalPrepaidAmount and subtract it from ram:DuePayableAmount (maybe adding an IncludedNote for clarification).
